proposals/0759-react-native-frameworks.md

```diff
@@ -0,0 +1,452 @@
+
+We consider the Framework and Platform concepts as **orthogonal**, i.e. you can use a Framework to ship to multiple platforms, and a Platform can be targeted by multiple Frameworks. In this context the role of a Framework is to abstract complexity and make it easier to ship to multiple Platforms providing a cohesive Developer Experience (see below).
+
+## Responsibilities
+
+This section describes where we draw the line between the React Native package and Frameworks.
+
+> [!NOTE]
+> For the sake of simplicity, we will be focusing on the React Native package (i.e. the NPM package called `react-native` which offers first-party support for Android & iOS).
+
+### What does the React Native package do
+
+The React Native package is responsible for **core APIs and capabilities** to build native apps. React Native is also responsible for offering **APIs** that allow Frameworks to deliver their experience. When such APIs are missing, it’s our interest to create them and have a conversation with Framework developers to inform their development, and help framework developers merge changes back into React Native.
+
+React Native offers first-party support for Android & iOS. Other platform support is offered by other libraries or Frameworks.
+
+Here is a (non-exhaustive) list of areas of responsibilities of the React Native package:
+
+- **Rendering & Components** - The React Native package is responsible for rendering on screen and offering basic building blocks to build custom UI. Parts of this API surface are:
+  - The Native Renderer API & Native Component API
+  - APIs of Core Components provided by React Native
+  - Accessibility support for such components
+- **Native Modules** - The React Native package is responsible for offering a mechanism and an API to access the underlying target platform. Parts of this API surface are:
+  - The Native Modules API
+  - APIs of Core Native Modules provided by React Native
+- **Runtime & Initialization** - The React Native package is responsible for offering an API to get access to the runtime and defining its lifecycle. The bridge and the bridgeless API are part of this API surface.
+- **Internals mechanism** - The React Native package is responsible of encapsulating away complexity and handling mechanisms such as:
+  - The threading model
+  - Interoperability layers between different architectures
+- **JS Engine Integration** - The React Native package is responsible for offering an API to integrate a target JS engine. React Native is distributed with built-in integrations for Hermes and JavaScriptCore. Third-party engines (like V8) may be integrated but are not directly supported.
+- **Bundler integration** - The React Native package defines the integration with Metro, the React Native bundler. Part of this API surface are:
+  - Building the JS and asset bundle
+  - The development server
+  - Debugging integrations
+  - Sourcemap support
+  - Error and Crash reporting
```

Comment:
we've discussed this before but this is definitely one of the areas that we'd love to be an extension point, so there is a public api that allows us swap out the redbox and other 'dev support' UI with alternatives (similar to how react-dom oriented frameworks provide their own error handling experiences that can be more framework-context-aware)

also related to this is the mechanism for loading a bundle. we often get bit by changes here and they are not the easiest to test for in CI, for example in react-native 0.73:
- https://github.com/expo/expo/pull/26630
- https://github.com/expo/expo/pull/26441

Reply:
> one of the areas that we'd love to be an extension point, so there is a public api that allows us swap out the redbox and other 'dev support' UI with alternatives

Yup I agree here.

I'd like to have @motiz88 @huntie @RobHogan to help with the framing as this falls under their domain.